tests/datasets/test_geo.py

```diff
@@ -84,6 +85,19 @@ def __len__(self) -> int:
         return 2
 
 
+@pytest.fixture(scope='module')
+def temp_archive(request: SubRequest) -> Generator[tuple[str, str], None, None]:
```

Comment:
@adamjstewart This temp archive fixture will run only once per input parameter (directory). The input comes from the test that uses it, hence the `indirect=True` in `pytest.mark.parametrize`.

Reply:
I've been looking for a way to do this for a long time, had no idea about `indirect=True`. Didn't expect to learn new pytest tricks while reviewing this PR, thanks!

Reply:
I figured I would try to do this properly ;)